java/src/hu/holyoil/controller/GameController.java

```diff
@@ -182,6 +193,184 @@ public void CheckGameCondition()  {
     public void StartGame()  {
         Logger.Log(this,"Starting game");
         // todo
+        // Generate between 100 and 500 asteroids
+        Random random = new Random();
+        int numOfAsteroids = 100;
+        if (Main.isRandomEnabled) {
+            numOfAsteroids = random.nextInt(401) + 100;
+        }
+
+        String startingAsteroidName = null;
+        String ufoStartingAsteroidName = null;
+
+        for (int i = 0; i < numOfAsteroids; i++) {
+
+            Asteroid asteroid = new Asteroid(
+                    AbstractBaseRepository.GetIdWithPrefix("Asteroid")
+            );
+
+            if (startingAsteroidName == null)
+                startingAsteroidName = asteroid.GetId();
+
+            ufoStartingAsteroidName = asteroid.GetId();
+
+            asteroid.SetNumOfLayersRemaining(
+                    Main.isRandomEnabled ? random.nextInt(6) : (i % 6)
+            );
+            asteroid.SetIsDiscovered(
+                    false
+            );
+            int generatedResource;
+            if (Main.isRandomEnabled) {
+
+                generatedResource = random.nextInt(5);
+
+            } else {
+
+                generatedResource = i / 20;
+
+            }
```

Comment:
`i / 20`??? Did you mean `i % 20`?

Reply:
This could be a one-liner:
```java
int generatedResource = Main.isRandomEnabled ? random.nextInt(5) : (i % 20);
```

Reply:
Why 20? Why not i % 5? (So the domain would be 0-4.)

Reply:
This method doesn't need to handle Math.isRandomEnabled, as it is purely for playing purposes. Not testing.

Reply:
Yes, the domain is currently 0-4 while it should be 0-5. I will fix that.